kura/test/org.eclipse.kura.driver.block.test/src/test/java/org/eclipse/kura/driver/block/test/BlockTaskAggregatorTest.java

```diff
@@ -0,0 +1,293 @@
+            if (inputBlocks.length % 2 != 0) {
+                fail("block list size must be a multiple of 2");
+            }
+            this.updateBlocks = updateBlocks;
+            return this;
+        }
+
+        public TestHelper expect(int... outputBlocks) {
+            if (outputBlocks.length % 2 != 0) {
+                fail("block list size must be a multiple of 2");
+            }
+            this.outputBlocks = outputBlocks;
+            return this;
+        }
+
+        public TestHelper prohibit(int... prohibitedBlocks) {
+            if (prohibitedBlocks.length % 2 != 0) {
+                fail("block list size must be a multiple of 2");
+            }
+            this.prohibitedBlocks = prohibitedBlocks;
+            return this;
+        }
+
+        public TestHelper setMinimumGapSize(int minimumGapSize) {
+            this.minimumGapSize = minimumGapSize;
+            return (this);
+        }
+
+        public TestHelper exec() {
+            List<Block> tasks = new ArrayList<Block>();
+            for (int i = 0; i < inputBlocks.length; i += 2) {
+                tasks.add(new TestTask(inputBlocks[i], inputBlocks[i + 1], taskListener));
+            }
+            for (int i = 0; i < updateBlocks.length; i += 2) {
+                tasks.add(new TestTask(updateBlocks[i], updateBlocks[i + 1], Mode.UPDATE, taskListener));
+            }
+
+            BlockTaskAggregator aggregator;
+
+            BlockFactory<ToplevelBlockTask> factory = (start, end) -> new TestTask(start, end, taskListener);
+
+            if (updateBlocks.length == 0) {
+                aggregator = new BlockTaskAggregator(tasks, factory);
+            } else {
+                aggregator = new UpdateBlockTaskAggregator(tasks, factory, factory);
+            }
+
+            aggregator.setMinimumGapSize(minimumGapSize);
+            for (int i = 0; i < prohibitedBlocks.length; i += 2) {
+                aggregator.addBlock(new ProhibitedBlock(prohibitedBlocks[i], prohibitedBlocks[i + 1]));
+            }
+
+            final Iterator<ToplevelBlockTask> blocks = aggregator.stream().iterator();
+
+            int childCount = 0;
+            if (outputBlocks != null) {
+                int i = 0;
+                while (blocks.hasNext()) {
+                    final ToplevelBlockTask next = blocks.next();
+                    assertEquals(outputBlocks[i * 2], next.getStart());
+                    assertEquals(outputBlocks[i * 2 + 1], next.getEnd());
+                    childCount += next.getChildren().size();
+                    try {
+                        next.run();
+                    } catch (IOException e) {
+                    }
+                    i++;
+                }
+                assertEquals(outputBlocks.length / 2, i);
+                assertEquals(inputBlocks.length / 2 + updateBlocks.length, childCount);
+            } else {
+                for (; blocks.hasNext(); blocks.next())
```

Comment:
This code was meant to be executed during the `shouldReportUnfeasibleProblem` tests, in this case it should be enough to trigger the aggregation process without checking anything until an exception is (or should be) thrown.
It is actually never executed because I forgot to remove the `.expect()` call, I will fix this.